presto-main/src/main/java/com/facebook/presto/execution/MemoryTrackingRemoteTaskFactory.java

```diff
@@ -74,10 +75,13 @@ public UpdatePeakMemory(QueryStateMachine stateMachine)
         @Override
         public synchronized void stateChanged(TaskStatus newStatus)
         {
-            long currentMemory = newStatus.getMemoryReservation().toBytes();
-            long deltaMemoryInBytes = currentMemory - previousMemory;
-            previousMemory = currentMemory;
-            stateMachine.updateMemoryUsage(deltaMemoryInBytes);
+            long currentUserMemory = newStatus.getMemoryReservation().toBytes();
+            long currentSystemMemory = newStatus.getSystemMemoryReservation().toBytes();
+            long deltaUserMemoryInBytes = currentUserMemory - previousUserMemory;
+            long deltaSystemMemoryInBytes = currentSystemMemory - previousSystemMemory;
+            previousUserMemory = currentUserMemory;
+            previousSystemMemory = currentSystemMemory;
+            stateMachine.updateMemoryUsage(deltaUserMemoryInBytes, deltaSystemMemoryInBytes);
```

Comment:
The existing code was unnecessarily roundabout. It can be simplified.

Here, we have the absolute number, and we turn it into delta. In `QueryStateMachine.updateMemoryUsage`, it takes the delta, and converts it back to absolute value.